sessionctx/variable/session.go

```diff
@@ -1281,6 +1281,68 @@ type SessionVars struct {
 	// OptPrefixIndexSingleScan indicates whether to do some optimizations to avoid double scan for prefix index.
 	// When set to true, `col is (not) null`(`col` is index prefix column) is regarded as index filter rather than table filter.
 	OptPrefixIndexSingleScan bool
+
+	MaxReuseChunk  int
+	MaxReuseColumn int
+	// ChunkPool Several chunks and columns are cached
+	ChunkPool struct {
+		Lock  sync.Mutex
+		Alloc chunk.Allocator
+	}
+	EnableReuseCheck bool
+
+	// UseChunkAlloc indicates whether statement use chunk alloc
+	UseChunkAlloc    bool
+	PreUseChunkAlloc bool
```

Comment:
Will the outside package change those field? if not, you can use lowercase like useChunkAlloc / perUseChunkAlloc etc

Reply:
UT test cases check the state of these variables. Need  change to function ?

Reply:
It's up to you.
Use lowercase and provide visitor function sometimes can help avoid the variables been modified unexpectedly, but make the code longer

Reply:
done